pennylane/compiler.py

```diff
@@ -0,0 +1,227 @@
+
+
+This subpackage exclusively functions as a wrapper for PennyLane's JIT compiler packages, without independently
+implementing any compiler itself. Currently, it supports the ``pennylane-catalyst`` package, with plans to
+incorporate additional packages in the near future.
+
+For any compiler packages seeking to be registered, it is imperative that they expose the 'entry_points' metadata
+under the designated group name: ``pennylane.compilers``.
+"""
+
+from typing import List
+from importlib import reload
+import pkg_resources
+
+
+class Compiler:
+    """The JIT compiler module containing all utilities and wrapper methods for ``qml.qjit`` in PennyLane."""
+
+    # Private class properties
+    __BACKENDS = []
+    __ENTRY_POINTS = {}
+
+    @classmethod
+    def available(cls, name: str = "pennylane-catalyst") -> bool:
+        """Check the availability of the given compiler package.
+
+        Args:
+            name (str): the name of the compiler package
+
+        Return:
+            bool : True if the compiler package is installed on the system
+        """
+
+        reload(pkg_resources)
+        pkg_name = pkg_resources.Requirement.parse(name)
+
+        if pkg_resources.working_set.find(pkg_name):
+            if name not in cls.__BACKENDS:
+                cls.__ENTRY_POINTS = {
+                    entry.name: entry
+                    for entry in pkg_resources.iter_entry_points("pennylane.compilers")
+                }
+                cls.__BACKENDS.append(name)
+            return True
+
+        if name in cls.__BACKENDS:
+            cls.__BACKENDS.remove(name)
+
+        return False
+
+    @classmethod
+    def available_backends(cls) -> List[str]:
+        """Return the available compiler packages."""
+        return cls.__BACKENDS
+
+    @classmethod
+    def active(cls) -> bool:
+        """Check whether the caller is inside a QJIT evaluation context.
+
+        Return:
+            bool : True if the caller is inside a QJIT evaluation context
+        """
+        if not cls.__BACKENDS:
+            raise RuntimeError("There is no available compiler package.")
+
+        if "cpl_utils" not in cls.__ENTRY_POINTS:
+            raise RuntimeError("There is no available 'cpl_utils' entry point.")
+
+        utils_loader = cls.__ENTRY_POINTS["cpl_utils"].load()
+        return utils_loader.contexts.EvaluationContext.is_tracing()
+
+    @classmethod
+    def qjit(cls, fn=None, *args, **kwargs):
+        """A just-in-time decorator for PennyLane and JAX programs using Catalyst.
+
+        This decorator enables both just-in-time and ahead-of-time compilation,
+        depending on whether function argument type hints are provided.
+
+        .. note::
+
+            Currently, ``lightning.qubit`` is the only supported backend device
+            for Catalyst compilation. For a list of supported operations, observables,
+            and measurements, please see the :doc:`/dev/quick_start`.
```

Comment:
This link is not showing in the docs

Reply:
I'm currently fixing the docs issues :+1: